java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -584,7 +653,18 @@ public static MessageMetadataResult readMessage(ReadChannel in) throws IOExcepti
     // Read the message size. There is an i32 little endian prefix.
     ByteBuffer buffer = ByteBuffer.allocate(4);
     if (in.readFully(buffer) == 4) {
-      int messageLength = MessageSerializer.bytesToInt(buffer.array());
+
+      int continuation = MessageSerializer.bytesToInt(buffer.array());
+      int messageLength = 0;
```

Comment:
I think this is for the back compatible purpose, new format contains "continues + length", but old format only contains "length", so we first try to read first 4 byte, if it equals to -1, it's new format, we then read length, otherwise is old format and length = first 4 bytes value.

Reply:
I mean could you just do this:

```java
int messageLength = MessageSerializer.bytesToInt(buffer.array());
if (messageLength == IPC_CONTINUATION_TOKEN) {
    buffer.clear();
    // ARROW-6313, if the first 4 bytes are continuation message, read the next 4 for the length
    if (in.readFully(buffer) != 4) {
        throw new IOException("Unexpected end of input trying to read message.");
    }
    messageLength = MessageSerializer.bytesToInt(buffer.array());
}
```

Reply:
I see, fixed.